samples/ProjectGrandmaster.Unity/Assets/AudioSystem.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+
+[RequireComponent(typeof(AudioSource))]
+public class AudioSystem : MonoBehaviour
+{
+    [SerializeField]
+    private AudioClip collisionSound;
+    private AudioSource source;
+    private bool isDropped;
+
+    private void Start()
+    {
+        isDropped = false;
+        source = this.GetComponent<AudioSource>();
+    }
+    private void OnTriggerEnter(Collider collision)
+    {
+        if (Time.frameCount < 100) return;
```

Comment:
Why is there a limit on when the event could start firing?

Reply:
When the game starts, the chess pieces drop onto the board. This causes all chess pieces to make the collision sound, hence limiting when the sound should be played.